### DISCUSSION TITLE
Existing natural gas generator

### DISCUSSION THREAD

**Discussion by lshaver** [NREL/REopt-API-Analysis#54] [7 - Technology Specific Questions]
I'm preparing to do an analysis for a site with an existing natural gas generator. On the input side, I believe can come up with equivalent figures for $/gal, gallons/kWh, and gallons/hour expressed in therms.

My concern is that the emissions rates for diesel and natural gas are different. Will the emissions components for which the diesel is responsible be included in the analysis, so that I can back these out and replace them with the equivalent natural gas emissions?

Are there other challenges to making this substitution that I'm not considering?

**Comment by adfarth** [comment]
As you reference, the assumed fuel for a generator in REopt is diesel fuel. However, you can modify the emissions rates of the diesel fuel (in this case, to reflect those of natural gas). If you're working in the web tool, click "Advanced Inputs" under the Renewable Energy & Emissions accordion. If a generator is selected under "Technologies" you should see a table for Fuel Emissions Factors. Here, you can modify the "Generator (diesel) fuel emissions factor (lb / gal)". 

The outputs page also separately reports emissions from on-site fuel burn and emissions from grid-purchased electricity, so a post-optimization substitution would be possible as well.
